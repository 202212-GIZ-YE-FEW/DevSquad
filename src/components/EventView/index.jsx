import React from "react";
import { AiFillClockCircle } from "react-icons/ai";
import { ImLocation } from "react-icons/im";
import EventImage from "../../components/EventImage/index";
import Buttoncomponent from "../Buttoncomponent";
import { useEffect, useState } from "react";
import { onAuthStateChanged } from "firebase/auth";
import { collection, getDocs, query, addDoc, where } from "firebase/firestore";
import { db, auth } from "../../../config/firebase";
import { useRouter } from "next/router";
export default function EventView(props) {
    const [userName, setuserName] = useState();
    const [isAuth, setIsAuth] = useState(null);
    // const [attendcount, setAttendcount] = useState();
    // const [userAttend, setUserAttend] = useState();
    const router = useRouter();
    onAuthStateChanged(auth, (user) => {
        user ? setIsAuth(auth?.currentUser?.email) : setIsAuth(null);
    });
    const usersCollectionRef = collection(db, "users");

    const getUserInfo = async (id) => {
        const q = query(usersCollectionRef, where("uid", "==", id));
        try {
            const data = await getDocs(q);
            const filteredData = data.docs.map((doc) => ({
                ...doc.data(),
                id: doc.id,
            }));

            // setuserName(filteredData[0].name);
            filteredData[0].name
                ? setuserName(filteredData[0].name)
                : setuserName(filteredData[0].email);
        } catch (err) {
            console.error(err);
        }
    };
    // const attendEvent = async (id) => {
    //     const userList = []
    //     const usersCollectionRef = collection(db, "users");
    //     const attendEventRef = collection(db, `events/${id}/attendEvent`);
    //     const dataAttend = await getDocs(attendEventRef);
    //     const data = dataAttend.docs.map((entry) => entry.data());

    //     setAttendcount(data.length)
    //     data.forEach(async (index) => {

    //         const users = await getDocs(query(usersCollectionRef, where('uid', "==", index.userId)))

    //         const userData = users.docs.map((index) => index.data());
    //         userData.forEach((index) => {
    //             userList.push(index)
    //         })
    //     })
    //     setUserAttend(userList)
    //     // console.log(userList);

    // };
    useEffect(() => {
        getUserInfo(props.entry.userId);
        // attendEvent(props.id)
    }, []);

    const joinEvent = async (id) => {
        try {
            const attendEventRef = collection(db, `events/${id}/attendEvent`);
            isAuth
                ? await addDoc(attendEventRef, {
                      userId: auth.currentUser.uid,
                  })
                : router.push("/signin");

            await addDoc(attendEventRef, {
                userId: auth.currentUser.uid,
            });

            alert("you are joined to the event");
        } catch (err) {
            console.error(err);
        }
    };

    return (
        <div className='md:grid grid-rows-2 gap-2 justify-center mt-8 md:gap-14 flex flex-col'>
            <div className='sm:grid sm:gap-4 sm:justify-center md:grid-cols-2 flex flex-col'>
                <div className='sm:m-0 sm:grid sm:justify-items-center md:flex md:col-span-2 m-4 order-2 sm:order-1'>
                    <h1 className='text-2xl font-medium font-Rubik'>
                        {/* The oragnization event name */}
                        {props.entry?.title}
                    </h1>
                </div>

                <div className='grid justify-items-center md:flex md:items-center sm:order-2'>
                    {/* <Image
                        src='/images/chart.png'
                        alt='chart'
                        width={500}
                        height={500}
                        className='w-5/6'
                    /> */}
                    <EventImage
                        pic={props.entry?.eventImage}
                        // src='/images/chart.png'
                        // alt='chart'
                        width='500px'
                        height='500px'
                        className='max-w-sm max-h-22'
                    />
                </div>

                {/* event info */}
                <div className='sm:grid sm:grid-cols-2 sm:justify-items-center md:flex md:flex-col order-last sm:order-3'>
                    <div className='md:space-y-4 sm:m-2 mb-4'>
                        {/* location */}
                        <div className='flex flex-row items-center'>
                            <ImLocation className='w-8' />
                            <p className='font-medium font-Rubik'>
                                {/* City, Country */}
                                {props.entry?.location}
                            </p>
                        </div>

                        {/* Date and place */}
                        <div className='flex flex-row items-center ml-8 md:ml-0'>
                            <AiFillClockCircle className='w-8 hidden md:block' />
                            <p className='text-gray-500 font-Rubik '>
                                {/* place of event and date */}
                                {props.entry?.eventDate}
                            </p>
                        </div>
                    </div>

                    <div className='space-y-4 m-2'>
                        {/* attendance */}
                        <div className='flex flex-row items-center'>
                            <div>
                                {/* <div class='relative inline-flex items-center justify-center w-8 h-8 bg-black rounded-full'>
                                    <span class='text-white font-Rubik'>R</span>
                                </div>
                                <div class='-left-4 relative inline-flex items-center justify-center w-8 h-8 bg-black rounded-full'>
                                    <span class='text-white font-Rubik'>R</span>
                                </div>
                                <div class='-left-8 relative inline-flex items-center justify-center w-8 h-8 bg-black rounded-full'>
                                    <span class='text-white font-Rubik'>R</span>
                                </div> */}
                            </div>
                            {/* <p className='font-Rubik'>+{attendcount} Attendance</p> */}
                            <p className='font-Rubik'>+12 Attendance</p>
                        </div>

                        {/* Organized by who */}
                        <div className='flex flex-row items-center'>
                            <div class='relative inline-flex items-center justify-center w-8 h-8 overflow-hidden bg-black rounded-full'>
                                <span class='s text-white font-Rubik'>
                                    {userName ? userName[0] : ""}
                                </span>
                            </div>
                            <p className='font-Rubik ml-2'>
                                Organized by {userName}
                            </p>
                        </div>
                    </div>

                    <div className='col-span-2 grid justify-items-center md:flex md:px-3'>
                        <Buttoncomponent
                            font='font-Rubik'
                            bgColor='bg-orange-400'
                            fontSize='text-sm'
                            height='h-7'
                            width='w-44 sm:w-52'
                            borderRaduis='rounded-md'
                            textColor='text-white'
                            margin='mt-5'
                            label='JOIN'
                            onClick={() => {
                                joinEvent(props.id);
                            }}
                        />
                    </div>
                </div>
            </div>
            {/* for the description and attendance */}
            <div className='sm:grid sm:gap-5 sm:justify-center md:grid-cols-2 md:m-0 flex flex-col m-2 space-y-4 sm:space-y-0'>
                <div>
                    <h1 className='font-Rubik font-medium text-lg'>
                        Event Description:
                    </h1>
                    <p className='font-Rubik text-gray-500 sm:max-w-md'>
                        {/* The oldest classical British and Latin writing had
                        little or no space between words and could be written in
                        boustrophedon alternating directions. Over time, text
                        direction left to right became standardized. dividers
                        and terminal punctuation became common. The first way to
                        divide sentences into groups wa original paragraphos,
                        similar to an underscore at the beginning of the new
                        group. The Greek par√°gr evolved into the pilcrow, which
                        in English manuscripts in the Middle Ages can be seen
                        inserted inline between sentences. */}
                        {props.entry?.description}
                    </p>
                </div>

                <div>
                    <p className='font-medium font-Rubik text-lg mb-2'>
                        Attendance:
                    </p>
                    {/* <div className='grid grid-cols-3 gap-2 sm:grid-cols-6 max-w-md md:w-80 md:grid-cols-4'>
                        <div className='flex flex-col items-center'>
                            <div class=' inline-flex items-center justify-center w-12 h-12 m-2 bg-black rounded-full'>
                                <span class='font-medium text-white'>R</span>
                            </div>
                            <p className='font-Rubik md:text-base text-sm'>
                                Jangis M.
                            </p>

                        </div>
                    </div> */}
                    {/* <div className='flex flex-row items-center'>
                        <div>
                            {userAttend &&
                                userAttend.map((index) => {
                                    return (
                                        <div class='-left-4 relative inline-flex items-center justify-center w-8 h-8 bg-black rounded-full'>
                                            <span class='text-white font-Rubik'>{index.name[0]}</span>
                                        </div>

                                    )
                                })
                            }

                        </div>
                    </div> */}
                </div>
            </div>
        </div>
    );
}
