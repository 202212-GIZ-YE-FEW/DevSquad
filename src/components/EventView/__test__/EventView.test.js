import renderer from "react-test-renderer";

import EventView from "../index";

it("renders correctly", () => {
    const tree = renderer.create(<EventView />).toJSON();
    expect(tree).toMatchSnapshot();
});

jest.mock("next/router", () => ({
    useRouter() {
        return {
            route: "/",
            pathname: "",
            query: "",
            asPath: "",
        };
    },
}));
