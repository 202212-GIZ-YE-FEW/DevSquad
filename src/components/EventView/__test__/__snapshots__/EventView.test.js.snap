// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="md:grid grid-rows-2 gap-2 justify-center mt-8 md:gap-14 flex flex-col"
>
  <div
    className="sm:grid sm:gap-4 sm:justify-center md:grid-cols-2 flex flex-col"
  >
    <div
      className="sm:m-0 sm:grid sm:justify-items-center md:flex md:col-span-2 m-4 order-2 sm:order-1"
    >
      <h1
        className="text-2xl font-medium font-Rubik"
      />
    </div>
    <div
      className="grid justify-items-center md:flex md:items-center sm:order-2"
    >
      <div>
        <img
          className="max-w-sm max-h-22"
          height="500px"
          src=""
          width="500px"
        />
      </div>
    </div>
    <div
      className="sm:grid sm:grid-cols-2 sm:justify-items-center md:flex md:flex-col order-last sm:order-3"
    >
      <div
        className="md:space-y-4 sm:m-2 mb-4"
      >
        <div
          className="flex flex-row items-center"
        >
          <svg
            className="w-8"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            version="1.1"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0c-2.761 0-5 2.239-5 5 0 5 5 11 5 11s5-6 5-11c0-2.761-2.239-5-5-5zM8 8c-1.657 0-3-1.343-3-3s1.343-3 3-3 3 1.343 3 3-1.343 3-3 3z"
            />
          </svg>
          <p
            className="font-medium font-Rubik"
          />
        </div>
        <div
          className="flex flex-row items-center ml-8 md:ml-0"
        >
          <svg
            className="w-8 hidden md:block"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm176.5 585.7l-28.6 39a7.99 7.99 0 0 1-11.2 1.7L483.3 569.8a7.92 7.92 0 0 1-3.3-6.5V288c0-4.4 3.6-8 8-8h48.1c4.4 0 8 3.6 8 8v247.5l142.6 103.1c3.6 2.5 4.4 7.5 1.8 11.1z"
            />
          </svg>
          <p
            className="text-gray-500 font-Rubik "
          />
        </div>
      </div>
      <div
        className="space-y-4 m-2"
      >
        <div
          className="flex flex-row items-center"
        >
          <div />
          <p
            className="font-Rubik"
          >
            +
             Attendance
          </p>
        </div>
        <div
          className="flex flex-row items-center"
        >
          <div
            class="relative inline-flex items-center justify-center w-8 h-8 overflow-hidden bg-black rounded-full"
          >
            <span
              class="s text-white font-Rubik"
            />
          </div>
          <p
            className="font-Rubik ml-2"
          >
            Organized by 
          </p>
        </div>
      </div>
      <div
        className="col-span-2 grid justify-items-center md:flex md:px-3"
      >
        <button
          className="bg-orange-400 text-white w-44 sm:w-52 h-7 font-Rubik text-sm  rounded-md undefined mt-5 p-2 "
          onClick={[Function]}
        >
          JOIN
        </button>
      </div>
    </div>
  </div>
  <div
    className="sm:grid sm:gap-5 sm:justify-center md:grid-cols-2 md:m-0 flex flex-col m-2 space-y-4 sm:space-y-0"
  >
    <div>
      <h1
        className="font-Rubik font-medium text-lg"
      >
        Event Description:
      </h1>
      <p
        className="font-Rubik text-gray-500 sm:max-w-md"
      />
    </div>
    <div>
      <p
        className="font-medium font-Rubik text-lg mb-2"
      >
        Attendance:
      </p>
    </div>
  </div>
</div>
`;
